fireplace-swing/src/main/java/io/github/bric3/fireplace/flamegraph/NodeDisplayStringProvider.java

```diff
@@ -0,0 +1,11 @@
+package io.github.bric3.fireplace.flamegraph;
+
+@FunctionalInterface
+public interface NodeDisplayStringProvider<T> {
+    /**
+     * @param node The node to calculate the displayed value for
+     * @param isRoot True if this is the root node in the graph
+     * @return The String to render for the frame
+     */
+    String frameString(T node, boolean isRoot);
```

Comment:
Used this instead of a BiFunction to avoid boxing

Reply:
Good call, I wonder if the API can be simplified to even avoid this boolean, trusting the actual node to know if it's a root node.

----
Thinking ahead we may want to draw additional decorations depending on the type of the frame (native, inlined, interpreted, ...). Which may lead to revise this contract once we get the requirements straight. Not there yet though.